astropy/modeling/core.py

```diff
@@ -1546,7 +1846,7 @@ def _check_param_broadcast(self, params, max_ndim):
                 "to the broadcasting rules.".format(param_a, shape_a,
                                                     param_b, shape_b))
 
-    def _param_sets(self, raw=False):
+    def _param_sets(self, raw=False, units=False):
```

Comment:
Adding units as a parameter somehow broke the shape of the returned array in the presence of units. For example

```
g1=models.Gaussian1D(10*u.A, 3*u.m, 1*u.m)
g1._param_sets(raw=True)
Out[20]: 
array([[ 10.],
       [  3.],
       [  1.]])

g1._param_sets(raw=True, units=True)
Out[19]: array([<Quantity [ 10.] A>, <Quantity [ 3.] m>, <Quantity [ 1.] m>], dtype=object)

```
This may seem insignificant but at some point it was important to have that shape. I can't tell right now without reading more whether this is still the case. However, I think we should try to be consistent.